### PR TITLE
Reorganize interceptors

### PR DESCRIPTION
**Pull request by fangluguomsft**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,117 +1,200 @@
...
 
-The logging policy logs details about each HTTP request and response, including the request method, URL, status code, and duration.
+ClientInterceptor. It propagates MD from incoming to outgoing. Only need to be used in servers. No need to be used in a pure client app.
 
-### Usage
-To use the logging policy, you need to create a logger and then apply the policy to your HTTP client.
+---
 
-Code example is included in the test code
+The following gRPC client interceptors are implemented by other open source projects. We enabled them in our default client interceptor list. [[return []grpc.UnaryClientInterceptor{](https://github.com/Azure/aks-middleware/blob/afd08e520d5d70f1b24910d26c2a686a0468feaa/interceptor/interceptor.go#L76-L77)
```
Comment (toma3233) [inline README.md:101]: the display text for this link should be fixed

File: README.md [hunk 2]
```diff
@@ -1,117 +1,200 @@
 <!-- markdownlint-disable MD004 -->
-# Overview
+
+# AKS middleware for Go
+
+<!-- vscode-markdown-toc -->
+* 1. [Usage](#Usage)
```
Comment (toma3233) [inline README.md:6]: Not sure if this is just an issue for me locally, but clicking the links does not take me to the section in the readme

File: README.md [hunk 3]
```diff
@@ -1,117 +1,200 @@
...
+
+### 2.1. <a name='requestid'></a>requestid
 
 ServerInterceptor. It adds x-request-id to MD if there is no such entry. This interceptor needs to be registered first so that its request-id can be used by autologger and applogger.
 
-# mdforward
+### 2.2. <a name='ctxloggerapplogger'></a>ctxlogger (applogger)
 
-ClientInterceptor. It propagates MD from incoming to outgoing. Only need to be used in servers. No need to be used in a pure client app.
+ServerInterceptor. It adds a logger to ctx with important information (e.g., request-id, method-name) already populated. App's handler code can get the logger and output additional information. Needs to be registered after the requestid interceptor.
```
Comment (toma3233) [inline README.md:59]: Since we are putting this under the gRPC server function, mentioning server interceptor here feels redundant

Comment (fangluguomsft) [reply README.md:59]: Done.

Comment (fangluguomsft) [reply README.md:59]: Done

File: README.md (hunk 2 above)
Comment (fangluguomsft) [reply README.md:6]: Fixed

**Review by toma3233 — APPROVED** [review]

**Pull request merged**